internal/core/operation.go

```diff
@@ -54,15 +56,15 @@ func (a *App) doOperation(
 	log hclog.Logger,
 	op operation,
 ) (interface{}, proto.Message, error) {
-	// Get our hooks
-	hooks := op.Hooks(a)
-
 	// Init the metadata
 	msg, err := op.Init(a)
 	if err != nil {
 		return nil, nil, err
 	}
 
+	// Get our hooks
+	hooks := op.Hooks(a)
```

Comment:
Did this need to move because of the added functionality to reinit components, or is this a "this should have been here all along" thing?